alchemy/src/stripe/tax-rate.ts

```diff
@@ -0,0 +1,198 @@
+   */
+  active?: boolean;
+  /**
+   * Two-letter country code (ISO 3166-1 alpha-2)
+   */
+  country?: string;
+  /**
+   * An arbitrary string attached to the tax rate for your internal use only
+   */
+  description?: string;
+  /**
+   * The jurisdiction for the tax rate
+   */
+  jurisdiction?: string;
+  /**
+   * Set of key-value pairs that you can attach to an object
+   */
+  metadata?: Record<string, string>;
+  /**
+   * ISO 3166-2 subdivision code, without country prefix
+   */
+  state?: string;
+  /**
+   * The high-level tax type, such as vat or sales_tax
+   */
+  taxType?: Stripe.TaxRateCreateParams.TaxType;
+}
+
+/**
+ * Output from the Stripe tax rate
+ */
+export interface TaxRate extends Resource<"stripe::TaxRate">, TaxRateProps {
+  /**
+   * The ID of the tax rate
+   */
+  id: string;
+  /**
+   * String representing the object's type
+   */
+  object: "tax_rate";
+  /**
+   * Time at which the object was created
+   */
+  created: number;
+  /**
+   * Has the value true if the object exists in live mode or the value false if the object exists in test mode
+   */
+  livemode: boolean;
+}
+
+/**
+ * Create and manage Stripe tax rates
+ *
+ * @example
+ * // Create a basic sales tax rate
+ * const salesTax = await TaxRate("ca-sales-tax", {
+ *   displayName: "CA Sales Tax",
+ *   percentage: 8.5,
+ *   inclusive: false,
+ *   active: true,
+ *   country: "US",
+ *   state: "CA",
+ *   description: "California sales tax",
+ *   taxType: "sales_tax"
+ * });
+ *
+ * @example
+ * // Create a VAT tax rate
+ * const vatTax = await TaxRate("uk-vat", {
+ *   displayName: "UK VAT",
+ *   percentage: 20.0,
+ *   inclusive: true,
+ *   active: true,
+ *   country: "GB",
+ *   description: "United Kingdom Value Added Tax",
+ *   taxType: "vat",
+ *   metadata: {
+ *     region: "europe",
+ *     tax_authority: "hmrc"
+ *   }
+ * });
+ *
+ * @example
+ * // Create a city tax rate
+ * const cityTax = await TaxRate("nyc-tax", {
+ *   displayName: "NYC Local Tax",
+ *   percentage: 4.5,
+ *   inclusive: false,
+ *   active: true,
+ *   country: "US",
+ *   state: "NY",
+ *   jurisdiction: "New York City",
+ *   description: "New York City local sales tax",
+ *   taxType: "sales_tax",
+ *   metadata: {
+ *     locality: "nyc",
+ *     tax_level: "city"
+ *   }
+ * });
+ */
+export const TaxRate = Resource(
+  "stripe::TaxRate",
+  async function (
+    this: Context<TaxRate>,
+    _id: string,
+    props: TaxRateProps,
+  ): Promise<TaxRate> {
+    const apiKey = process.env.STRIPE_API_KEY;
+    if (!apiKey) {
+      throw new Error("STRIPE_API_KEY environment variable is required");
+    }
+
+    const stripe = new Stripe(apiKey);
+
+    if (this.phase === "delete") {
+      try {
+        if (this.output?.id) {
+          await stripe.taxRates.update(this.output.id, { active: false });
+        }
+      } catch (error) {
+        console.error("Error deactivating tax rate:", error);
+      }
```

Comment:
do not swallow errors. that is bad. only error that can be swallowed during a deletion is Not Found. go look up stripe api docs and make sure you detect that properly